src/app/auth/auth-registration/auth-registration.component.ts

```diff
@@ -0,0 +1,259 @@
+        Validators.required
+      ])
+    });
+  }
+
+  validateQueryParams() {
+    // access query params
+    this.route.queryParamMap.subscribe(queryParams => {
+      let email = queryParams.get("email");
+      this.user.email = email;
+      let activation_code = queryParams.get("activation_code");
+      if (email || activation_code) {
+        // set query params to stroage
+        this.storage.set("hash", {
+          email: email,
+          key: activation_code
+        });
+        // check is Url valid or not.
+        this.authService
+          .verifyRegistration({
+            email: this.storage.get("hash").email,
+            key: this.storage.get("hash").key
+          })
+          .subscribe(
+            response => {
+              let data = response.data.data;
+              let user = data.user;
+              let hash = this.storage.get("hash");
+              this.user.email = email;
+              this.user.contact = (data.User || {}).contact_number || null;
+              let merge = { hash, user };
+              this.storage.set("hash", merge);
+
+              // get app configaration
+              this.authService
+                .checkDomain({
+                  domain: this.domain
+                })
+                .subscribe(
+                  response => {
+                    var data = (response.data || {}).data;
+                    data = this.utils.find(data, function(datum) {
+                      return (
+                        datum.config && datum.config.auth_via_contact_number
+                      );
+                    });
+
+                    if (data && data.config) {
+                      if (data.config.auth_via_contact_number === true) {
+                        this.hide_password = true;
+                        this.user.password = this.autoGeneratePassword();
+                        this.confirmPassword = this.user.password;
+                      }
+                    }
+                  },
+                  err => {}
+                );
+            },
+            error => {
+              this.utils.popUp(
+                "shortMessage",
+                {
+                  message: "Registration link invalid"
+                },
+                false
+              );
+            }
+          );
+      } else {
+        this.utils.popUp(
+          "shortMessage",
+          {
+            message: "Registration link invalid"
+          },
+          false
+        );
+      }
+    });
+  }
+
+  private autoGeneratePassword() {
+    let text = Md5.hashStr("").toString();
+    let autoPass = text.substr(0, 8);
+    return autoPass;
+  }
+
+  openLink() {
+    window.open(
+      "https://images.practera.com/terms_and_conditions/practera_default_terms_conditions_july2018.pdf",
+      "_system"
+    );
+  }
+
+  register() {
+    this.validateRegistration();
+    if ((this.registerationForm.valid) && (this.errors.length === 0)) {
```

Comment:
Since you already have a function to do the validation, `this.validateRegistration()` should return a boolean and be used here as the condition.